src/steal.c

```diff
@@ -506,82 +506,51 @@ register struct obj *otmp;
     return freed_otmp;
 }
 
-/* called for AD_SAMU (the Wizard and quest nemeses) */
 void
-stealamulet(mtmp)
+stealamulet(mtmp, allow_quest_artifact)
 struct monst *mtmp;
+boolean allow_quest_artifact;
 {
-    char buf[BUFSZ];
-    struct obj *otmp = 0, *obj = 0;
-    int real = 0, fake = 0, n;
-
-    /* target every quest artifact, not just current role's;
-       if hero has more than one, choose randomly so that player
-       can't use inventory ordering to influence the theft */
-    for (n = 0, obj = invent; obj; obj = obj->nobj)
-        if (any_quest_artifact(obj))
-            ++n, otmp = obj;
-    if (n > 1) {
-        n = rnd(n);
+    struct obj *otmp = (struct obj *) 0;
+    int real = 0, fake = 0;
+
+    /* select the artifact to steal */
+    if (u.uhave.amulet) {
+        real = AMULET_OF_YENDOR;
+        fake = FAKE_AMULET_OF_YENDOR;
+    } else if (u.uhave.questart && allow_quest_artifact) {
```

Comment:
The check on u.uhave.questart will prevent the Wizard from stealing other artifacts if you don't have your own.